### PR TITLE
[litehtml] new port version 0.6

### PR DESCRIPTION
**Pull request by rioki**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See docs/examples/adding-an-explicit-usage.md for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Neumann-A** [comment]
Note: litehtml is vendored by qttools currently. I don't know if it is visible from the outside or just a part of qtassistent.

**Comment by rioki** [comment]
> Note: litehtml is vendored by qttools currently. I don't know if it is visible from the outside or just a part of qtassistent.

As far as I can see the QT version runs as qlitehtml and has some changes to the code. It's API and ABI incompatible as it uses QT specific types and should be considered a sperate fork.

If I see it correctly qtassistent consumes qlitehtml as a [static library and never installs anything](https://github.com/qt/qttools/blob/dev/src/assistant/CMakeLists.txt#L25).

**Comment by dg0yt** [comment]
qlitehtml is a wrapper and needs litehtml. It will use an installed version if it is found:
https://code.qt.io/cgit/playground/qlitehtml.git/tree/src/CMakeLists.txt#n18

So AFAICT qttools must use the vcpkg version of litehtml if we don't want to have two providers of the same symbols.
Another questions is if we also want/need a qlitehtml port...

**Comment by Neumann-A** [comment]
> Another questions is if we also want/need a qlitehtml port...

I don't think that is currently needed.

**Comment by jimwang118** [comment]
All test passed with following triplets:
x86-windows
x64-windows
x64-windows-static

File: ports/litehtml/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,25 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO litehtml/litehtml
+    REF v0.6
```
Comment (JonLiu1993) [inline ports/litehtml/portfile.cmake:4]:
```suggestion
    REF "v${VERSION}"
```

File: ports/litehtml/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/litehtml)
+vcpkg_fixup_pkgconfig()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (JonLiu1993) [inline ports/litehtml/portfile.cmake:25]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by dg0yt** [comment]
IIUC litehtml still needs to be devendored from qttools.

**Comment by rioki** [comment]
I will see if I can devendor litehtml this weekend.

**Comment by rioki** [comment]
I fixed the CMakeLists.txt in qlitehtml/src so that is used the vcpkg litehtml. git still pulls the submodule, but I have no idea how to fix that, since I can patch the .gitmodules but that happens only after the submodule was pulled.

**Comment by rioki** [comment]
Not sure why this fails on the CI but not on my machine. Will make gumbo static liked and patch the CMake config, as gumbo-parser is vendored into litehtml and the official repo is marked as defunct.

**Comment by rioki** [comment]
Looking into why it fails on the CI (works on my machine™)

**Comment by rioki** [comment]
I fixed litehtml exposing the vendored [gumbo](https://github.com/google/gumbo-parser) for CMake to handle it's dependencies cleanly. The project is defunct, thus I don't think making a unique port is needed.

**Review by jimwang118 — APPROVED** [review]

**Review by jimwang118 — APPROVED** [review]

**Comment by dg0yt** [comment]
> I fixed litehtml exposing the vendored [gumbo](https://github.com/google/gumbo-parser) for CMake to handle it's dependencies cleanly. The project is defunct, thus I don't think making a unique port is needed.

There is a gumbo port, and we have a file conflict in CI now.
https://dev.azure.com/vcpkg/public/_build/results?buildId=85761&view=logs&j=84da4f4a-968a-5b5c-16e9-f444c845396d&t=8cdf6620-a129-5b70-3f3e-822630a64446

**Comment by rioki** [comment]
Sorry for missing that. I will look into fixing this.

**Pull request merged**